public/app/features/templating/query_variable.ts

```diff
@@ -139,8 +140,8 @@ export class QueryVariable implements Variable {
     });
   }
 
-  metricFindQuery(datasource: any, query: string) {
-    const options: any = { range: undefined, variable: this };
+  metricFindQuery(datasource: any, query: string, searchFilter?: string) {
+    const options: any = { range: undefined, variable: this, searchFilter };
```

Comment:
Is the `searchFilter` option something that's available for every datasource?

Reply:
available in the sense that it can be implemented, but it needs to be handled in the DataSourceApi metricFindQuery function

Reply:
I've added support for Graphite, MySQL, PostGres and TestData